extensions-contrib/kafka-segment-metadata-emitter/src/main/java/org/apache/druid/emitter/kafkasegmentmetadata/KafkaSegmentMetadataEmitter.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.kafkasegmentmetadata;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Timestamp;
+import io.confluent.streaming.aggregates.proto.DruidSegmentEvent;
+import org.apache.druid.java.util.common.lifecycle.LifecycleStop;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.SegmentMetadataEvent;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.joda.time.DateTime;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class KafkaSegmentMetadataEmitter implements Emitter
+{
+
+  private static Logger log = new Logger(KafkaSegmentMetadataEmitter.class);
+
+  private static final int DEFAULT_RETRIES = 3;
+  private final AtomicLong segmentMetadataLost;
+
+  private final KafkaSegmentMetadataEmitterConfig config;
+  private final Producer<String, DruidSegmentEvent> producer;
+  private final ObjectMapper jsonMapper;
+  private final ScheduledExecutorService scheduler;
+
+  public KafkaSegmentMetadataEmitter(KafkaSegmentMetadataEmitterConfig config, ObjectMapper jsonMapper)
```

Comment:
why do we need a different kind of kafka emitter for this particular event? This makes it hard to maintain since we would now have to maintain and apply patches to multiple kafka emitters. It would probably make more sense to reuse the existing kafka event emitter and add the ability to choose which types of event / which format to send them in.

Reply:
As discussed, I also feel it may be better to model this new data as an event and use the same kafka emitter to emit the data. Currently the kafka emitter has only metricsQueue and alertQueue, may be we add eventsQueue and push this data into eventsQueue. If finer control is required, we can enhance the config to control which data (metrics/alerts/events) goes into the kafka topic. @harinirajendran - wdyt ?

Reply:
I'll think on this and update the PR.

Reply:
Updated the PR by moving the logic of segment metadata emission into kafka emitter